web/src/ui_util.ts

```diff
@@ -44,6 +44,62 @@ export function change_katex_to_raw_latex($element: JQuery): void {
     });
 }
 
+export function is_user_said_paragraph($element: JQuery): boolean {
+    // Irrespective of language, the user said paragraph has these exact elements:
+    // 1. A user mention
+    // 2. A message link ("said")
+    // 3. A colon
+    const $user_mention = $element.find(".user-mention");
+    if ($user_mention.length !== 1) {
+        return false;
+    }
+    // A message link in the same organisation always starts with `#narrow/` and ends with
+    // `/near/<message_id>`, where <message_id> is a number.
+    const $message_link = $element.find("a[href^='#narrow/']").filter((_index, element) => {
+        const href = $(element).attr("href");
+        return href ? /\/near\/\d+$/.test(href) : false;
+    });
```

Comment:
Let's have a proper test for whether it's the right kind of URL in a function -- define a `hash_util.is_conversation_and_time_url`, and call that in this block rather than doing a direct regex here. Some `hash_parser` code may help.

(Localizing the key part of this logic there will help make sure we update this code if/when we change the URL formats)

Reply:
Importing `hash_util` into `ui_util` is causing a dependency cycle :confused: Defining the function in `hash_parser` and importing that into `ui_util` works, but I am not sure that's the ideal place for the function

Reply:
Putting it in `hash_parser` sounds fine.

Reply:
@timabbott Done with tests! I've also made a prep commit with a small refactor that helped this new function be cleaner